src/hazelcore/semantics/CursorInfo.re

```diff
@@ -946,6 +1049,101 @@ let ana_cursor_found_exp =
     );
   };
 
+let rec find_uses_block =
+        (x: Var.t, block: UHExp.block, steps, index): uses_list => {
+  let UHExp.Block(lines, e) = block;
+  let (uses, shadow) = find_uses_lines(x, lines, steps @ [0], index, false);
+  shadow
+    ? uses
+    : uses @ find_uses_exp(x, e, steps @ [index + List.length(lines)]);
+}
+and find_uses_lines =
+    (x: Var.t, lines: UHExp.lines, steps, index, shadow): (uses_list, bool) =>
+  switch (lines, shadow) {
```

Comment:
instead of pattern matching on the pair, just do an `if (shadowed) { ([], shadowed) } else { ... }`